### PR TITLE
Feature/pype 785 osx compatibility

### PR DESCRIPTION
**Pull request by antirotor**
## Basic macOS compatibility

This PR adds pype basic compatibility.

### Aditional dependecies:

It needs **Xcode Command Line Tools** installed (they are including git, gcc, etc.). Pype is trying to install them by itself but needs to be online.

| ❗ Tested on macOS 10.15.3 Catalina |
| --- |

Needs corresponding PR in **pype** and **pype-config**
Closes #21
Depends on pypeclub/pype#141
Depends on pypeclub/pype-config#24

### Know problems

Tray menu is rendered twice over itself. Native style seems to be overlaid by Qt styled menu.

### CODE REVIEW

File: pypeapp/lib/log.py [hunk 1]
```diff
@@ -51,7 +51,10 @@
 
 system_name, pc_name = platform.uname()[:2]
 host_name = socket.gethostname()
-host_id = socket.gethostbyname(host_name)
+try:
+    ip = socket.gethostbyname(host_name)
+except socket.gaierror:
+    ip = "127.0.0.1"
```
Comment (iLLiCiTiT) [inline pypeapp/lib/log.py:57]: Just note: I think host IP was added by me for more information in log records but I don't think It's required and may be removed....

File: deploy/deploy.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "repositories": [{
       "name": "avalon-core",
       "url": "https://github.com/pypeclub/avalon-core.git",
-      "branch": "2.7.0-pype"
+      "tag": "2.8.0-pype"
```
Comment (iLLiCiTiT) [inline deploy/deploy.json:7]: Shall we change this to develop? Because for deployment in clients we use their deployment json.

Comment (mkolar) [reply deploy/deploy.json:7]: no I would keep this locked to stable versions. Think of people who just wanna give it a shot.

**Comment by iLLiCiTiT** [comment]
PR contain changes from PR https://github.com/pypeclub/pype-setup/pull/21

**Comment by antirotor** [comment]
> PR contain changes from PR #21

As it is stated in description 🤖

**Comment by iLLiCiTiT** [comment]
> As it is stated in description 🤖

BTW: was the issue with permissions for ffmpeg binaries solved?

**Comment by antirotor** [comment]
> BTW: was the issue with permissions for ffmpeg binaries solved?

nope. As it affects all deployments (not only macOS issue) I'll make separate PR for this. Meanwhile it is sufficient to run:

```sh
sudo chmod +x ffmpeg
sudo chmod +x ffprobe
sudo chmod +x ffplay
```
in `vendor/bin/ffmpeg_exec/darwin/bin`.

**Pull request merged**